gobblin-compaction/src/main/java/org/apache/gobblin/compaction/mapreduce/orc/OrcUtils.java

```diff
@@ -491,18 +491,34 @@ public static WritableComparable createValueRecursively(TypeDescription schema)
    */
   public static boolean eligibleForUpConvertHelper(TypeDescription originalSchema, TypeDescription targetSchema) {
     if (!targetSchema.getCategory().isPrimitive()) {
-      if (!originalSchema.getFieldNames().containsAll(targetSchema.getFieldNames())) {
-        return false;
-      }
-      boolean result = true;
+      if (targetSchema.getCategory().equals(TypeDescription.Category.LIST)) {
+        return eligibleForUpConvertHelper(originalSchema.getChildren().get(0), targetSchema.getChildren().get(0));
```

Comment:
Should we do a null check here? What happens if originalSchema.getChildren() returns null?

Reply:
Will add a safety-guard.